libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/DirectionsSession.kt

```diff
@@ -1,15 +1,23 @@
 package com.mapbox.navigation.core.directions.session
 
+import android.location.Location
 import com.mapbox.api.directions.v5.models.DirectionsRoute
 import com.mapbox.api.directions.v5.models.RouteOptions
 import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.trip.model.RouteProgress
 
 internal interface DirectionsSession {
 
     var routes: List<DirectionsRoute>
 
     fun getRouteOptions(): RouteOptions?
 
+    fun buildAdjustedRouteOptions(
```

Comment:
`DirectionsSession` is potentially going to be available in the future so I'm wondering if we want to force clients to implement this API or if we should only add it to the default implementation `MapboxDirectionsSession` 🤔 What do you think?

Reply:
Hmm well MapboxNavigation is built to work with the `DirectionsSession` interface, so a client would need to override this too.

[When doing this](https://github.com/mapbox/mapbox-navigation-android/pull/2577/files#r391327640), my thought was it could use its own dedicated class. But `buildAdjustedRouteOptions` is similar to `getRouteOptions`. If I'm overriding getRouteOptions, seems like I should also override the route options during the ride..

Renaming this to `getAdjustedRouteOptions`

> if we want to force clients to implement this API or if we should only add it to the default implementation

provide the default implementation, and let the customer override it?

Reply:
Spoke offline, moved out to a new class instead of DirectionsSession